packages/eslint-plugin/src/rules/strict-void-return.ts

```diff
@@ -0,0 +1,991 @@
+
+      nonVoidReturnInImplMember:
+        'Value returned in `{{memberName}}` method of `{{className}}`, whose interface `{{baseName}}` declares it as a void method.',
+      asyncFuncInImplMember:
+        'Async function provided as `{{memberName}}` method of `{{className}}`, whose interface `{{baseName}}` declares it as a void method.',
+      asyncNoTryCatchFuncInImplMember:
+        'Async function not wrapped with a try-catch block and provided as `{{memberName}}` method of `{{className}}`, whose interface `{{baseName}}` declares it as a void method.',
+      genFuncInImplMember:
+        'Generator function provided as `{{memberName}}` method of `{{className}}`, whose interface `{{baseName}}` declares it as a void method.',
+      nonVoidFuncInImplMember:
+        'Value-returning function provided as `{{memberName}}` method of `{{className}}`, whose interface `{{baseName}}` declares it as a void method.',
+
+      nonVoidReturnInOther:
+        'Value returned in a context where a void return was expected.',
+      asyncFuncInOther:
+        'Async function used in a context where a void function was expected.',
+      asyncNoTryCatchFuncInOther:
+        'Async function not wrapped with a try-catch block and used in a context where a void function was expected.',
+      genFuncInOther:
+        'Generator function used in a context where a void function was expected.',
+      nonVoidFuncInOther:
+        'Value-returning function used in a context where a void function was expected.',
+
+      suggestWrapInAsyncIIFE:
+        'Wrap the function body in an immediately-invoked async function expression.',
+      suggestWrapInTryCatch: 'Wrap the function body in a try-catch block.',
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          considerOtherOverloads: { type: 'boolean' },
+          considerBaseClass: { type: 'boolean' },
+          considerImplementedInterfaces: { type: 'boolean' },
+          allowReturnPromiseIfTryCatch: { type: 'boolean' },
+          allowReturnUndefined: { type: 'boolean' },
+          allowReturnNull: { type: 'boolean' },
+          allowReturnAny: { type: 'boolean' },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  defaultOptions: [
+    {
+      considerOtherOverloads: true,
+      considerBaseClass: true,
+      considerImplementedInterfaces: true,
+      allowReturnPromiseIfTryCatch: true,
+      allowReturnUndefined: true,
+      allowReturnNull: true,
+    },
+  ],
+
+  create(context, [options]) {
+    const sourceCode = context.sourceCode;
+    const parserServices = util.getParserServices(context);
+    const checker = parserServices.program.getTypeChecker();
+
+    return {
+      'CallExpression, NewExpression': (
+        node: TSESTree.CallExpression | TSESTree.NewExpression,
+      ): void => {
+        checkFunctionCallNode(node);
+      },
```

Comment:
[Style] Nit: these can be inlined. Anything that just curries a single param ✨ 

```suggestion
      'CallExpression, NewExpression': checkFunctionCallNode,
```